contracts/core/EntryPoint.sol

```diff
@@ -313,8 +318,7 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
     ) internal pure {
         mUserOp.sender = userOp.sender;
```

Comment:
Why not have the entire `UserOperation` => `MemoryUserOp` conversion in the `UserOperationLib` library? Construction of a `MemoryUserOp` takes a lot of space and will call to the `UserOperationLib` at least twice.

Reply:
This copy is an internal optimization of EntryPoint. we might even remove copying some fields if we find they are not needed (e.g. used exactly once)
UserOperationLib is for general-use functions, that are expected to be used even outside the EntryPoint itself.